elements/pfe-progress-steps/src/pfe-progress-steps-item.js

```diff
@@ -0,0 +1,168 @@
+    }
+    if (this.state === "error") {
+      return `<svg height="100%" width="100%" viewBox="0 0 512 512" aria-hidden="true" role="img" aria-describedby="pf-tooltip-196" xmlns="http://www.w3.org/2000/svg"><path d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"></path></svg>`;
+    }
+    return ``;
+  }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Content;
+  }
+
+  static get properties() {
+    return {
+      state: {
+        type: String,
+        default: "inactive",
+        values: ["inactive", "done", "error"],
+        observer: "_build"
+      },
+      vertical: {
+        type: String,
+        default: false
+      },
+      current: {
+        type: Boolean,
+        default: false,
+        observer: "_build"
+      }
+    };
+  }
+
+  static get slots() {
+    return {
+      title: {
+        title: "Title",
+        type: "array",
+        namedSlot: true,
+        maxItems: 1,
+        items: {
+          $ref: "raw"
+        }
+      },
+      description: {
+        title: "Description",
+        type: "array",
+        namedSlot: true,
+        items: {
+          $ref: "raw"
+        }
+      }
+    };
+  }
+
+  constructor() {
+    super(PfeProgressStepsItem, {
+      type: PfeProgressStepsItem.PfeType
+    });
+    // programatically generate a link based on slot
+    this._hasLink = false;
+    // programatically skip links based on state
+    this._skipLink = false;
+
+    this.addEventListener("click", this._clickHandler.bind(this));
+    this.addEventListener("keydown", this._keydownHandler.bind(this));
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._build();
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener("click", this._clickHandler.bind(this));
+    this.removeEventListener("keydown", this._keydownHandler.bind(this));
+  }
+
+  _build() {
+    if (this.isIE11) return;
+
+    // decide if we should add aria-current
+    if (this.current) {
+      this.setAttribute("aria-current", "true");
+    } else {
+      this.removeAttribute("aria-current");
+    }
+
+    // find out if we should skip the link
+    this._skipLink = this.current || this.state === "error";
+
+    // Find out if there are any links
+    const link = this.querySelector(`a[slot="link"]`);
```

Comment:
What is the relationship between this and the description?

Reply:
The slot is basically being used as an alternative to creating a new property on this item called "src" or "link" or "href".  I was in the "lightdom first" headspace when I did this.